tests/feature_elimination/test_feature_elimination.py

```diff
@@ -248,3 +250,106 @@ def test_complex_dataset(complex_data, complex_lightgbm):
     assert report.shape[0] == X.shape[1]
 
     assert len(record) >= 2
+
+
+@pytest.mark.skipif(os.environ.get("SKIP_LIGHTGBM") == "true", reason="LightGBM tests disabled")
+def test_shap_rfe_early_stopping(complex_data, complex_lightgbm, capsys):
+    """
+    Test EarlyStoppingShapRFECV with a LGBMClassifier.
+    """
+    clf = complex_lightgbm
```

Comment:
I think it is better to not use this fixture, but rather initialize your own lgbm with 200 trees and max depth of 3, so you actually cause early stopping to kick in.